### PR TITLE
Webservice product az

### PR DESCRIPTION
**Pull request by thomasnares**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Add a guide to illustrate how to create and manage a product from A to Z with webservices

Needs https://github.com/PrestaShop/webservice-postman-examples/pull/4 to be merged

### CODE REVIEW

File: webservice/tutorials/create-product-az.md
Comment (thomasnares) [inline webservice/tutorials/create-product-az.md]:
```suggestion
[Download the Postman Collection](https://github.com/PrestaShop/webservice-postman-examples/blob/main/demo_product_az.json)
```

File: webservice/tutorials/create-product-az.md
Comment (thomasnares) [inline webservice/tutorials/create-product-az.md]:
```suggestion
title: Create a product from A to Z with the Webservices
```

**Review by matthieu-rolland — APPROVED** [review]
that's a nice addition 👍 

It looks good to me, I'd like at least a second opinion though before merging, maybe someone more familiar than I am with the legacy API

**Comment by matthieu-rolland** [comment]
I merged your json example

**Review by kpodemski — CHANGES_REQUESTED** [review]
I'm blocking, so I can review it before it gets merged 😁

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
title: Create a product from start to finish with Webservices
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
menuTitle: Creating a product - complete guide
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
# Create a product from start to finish with Webservices
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
This guide will demonstrate how to create and manipulate a `Product` and its dependencies from the Webservices.

Firstly, we will generate a "Supplier" and a "Manufacturer". Secondly, we will establish a "Category" to organize our "Product".
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Once the `Product` is added, we'll include some `Stocks`, upload an `Image`, and handle multi-language content.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
We will illustrate those examples [with a Postman Collection]({{<relref "/8/webservice/tutorials/testing-webservice-postman">}}), that you can download and run against your own PrestaShop Webservices. 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Although optional, having both `Manufacturer` and `Supplier` information will assist in managing your stock and business efficiently.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send a `POST` request to `/api/manufacturers` with the following body: 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send a `POST` request to `/api/suppliers` with the following body: 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Let's create a designated `Category` for our `Product`.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send a `POST` request to `/api/categories` with the following body:
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md:149]:
```suggestion
Ensure the provided `Language` ID corresponds to your `Language` ID in your back office.

Retrieve the newly created category `ID` from the Webservices's response.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Now that the required dependencies are in place, we can move forward with crafting the minimum viable product.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send a `POST` request to `/api/products` with the following body: 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
This will create a `Product`, of type: `ProductType::TYPE_STANDARD`, associated with the `Shop` with `id:1`.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Whenever a `Product` entity is formed, a corresponding `Stock_Available` record is also generated and inserted to the database.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
## Product Stock management
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
You can update the available quantities by sending a `PATCH` request to `/api/stock_availables/{{id}}` (with `{{id}}` the previously retrieved stock_available entity id). 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Depending on your stock management strategy (`Share available quantities for sale` between shops, in the `Back office > Advanced Parameters > Multistore > Edit shop group`), you need to provide an `id_shop` (per shop stock) or an `id_shop_group` (shared shop stock) to the request.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
## Product images management
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
To upload an `Image` to your `Product`, send a `POST` request to `/api/images/products/{{product_id}}` (with `{{id}}` of the newly created `Product` entity). 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
If you have two languages set up in your PrestaShop instance, for example:
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
## Create a Product with Combinations - ProductType::TYPE_COMBINATIONS
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
To create a Product with Combinations, you must make your `Product` and then add your `Combinations`. 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Change XML body of the [Product creation request]({{< relref "#create-the-product" >}}). Modify `<product_type>` node, and replace `standard` with `combinations`.
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send a `POST` request to `/api/products` with the following body: 
```

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [inline webservice/tutorials/create-product-az.md]:
```suggestion
Send two `POST` requests to `/api/combinations` to create your combinations. Use `{{id_product}}` of your previously created `Product`. 
```

Why do we need to send it twice? Also, it seems there's no information about the combination, like... which group of the attribute is thta? what's the value?

File: webservice/tutorials/testing-webservice-postman.md
Comment (kpodemski) [inline webservice/tutorials/testing-webservice-postman.md]:
```suggestion
## Create a product from start to finish with Webservices (demo_product_creation_AZ.json)
```

File: webservice/tutorials/testing-webservice-postman.md
Comment (kpodemski) [inline webservice/tutorials/testing-webservice-postman.md]:
```suggestion
This collection illustrates how to create a product from A to Z with all of its relations (to Brand, Category, Manufacturer, Images, etc.)
```

**Review by kpodemski — CHANGES_REQUESTED** [review]
Well done @thomasnares 👏🏻 

I left some suggestions and one question regarding managing products with combinations.

**Comment by LouisAUTHIE** [comment]
Nice one, ok with @kpodemski reviews. Looking forward to the next correction commit and it should be perfect @thomasnares . Thx

File: webservice/tutorials/create-product-az.md
Comment (kpodemski) [reply webservice/tutorials/create-product-az.md]: cc @thomasnares

Comment (thomasnares) [reply webservice/tutorials/create-product-az.md]: yes i still need to change this paragraph and explain about attributes 👍

Comment (kpodemski) [reply webservice/tutorials/create-product-az.md]: ok, I converted your PR to draft :D let me know once it's ready

**Comment by thomasnares** [comment]
I added a sction about managing Features & Feature Values, 

And improved the Combination example with Attributes

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Thanks @thomasnares ! This improves WS documentation a lot! well done 👏🏻

**Pull request merged**
